synapse/lib/oauth.py

```diff
@@ -73,6 +34,18 @@ def normOAuthTokenData(issued_at, data):
         'refresh_token': data.get('refresh_token'),
     }
 
+az_tfile_envar = 'AZURE_FEDERATED_TOKEN_FILE'
+def _getAzureTokenFile():
+    fp = os.getenv(az_tfile_envar, None)
+    if not fp:
+        return False, {'error': f'{az_tfile_envar} environment variable is not set.'}
+    if os.path.exists(fp):
+        with open(fp, 'r') as fd:
+            assertion = fd.read()
+            return True, {'token': assertion}
```

Comment:
Do we need this enveloping? Seems like the API could just return `True, assertion`?

Reply:
Was trying to keep it as `tuple[bool, dict]` call signature.

Reply:
changed this to `tuple[bool, str]`